watcher/local/registry.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"sync"
+
+	"perun.network/go-perun/channel"
+)
+
+type (
+	registry struct {
+		mtx sync.Mutex
+		chs map[channel.ID]*ch
+	}
+)
+
+func newRegistry() *registry {
+	return &registry{
+		chs: make(map[channel.ID]*ch),
+	}
+}
+
+// lock locks the registry.
+func (r *registry) lock() {
+	r.mtx.Lock()
+}
+
+// unlock unlocks the registry.
+func (r *registry) unlock() {
+	r.mtx.Unlock()
+}
+
+// addUnsafe adds the channel to registry, expecting the caller to have locked
+// the registry.
+func (r *registry) addUnsafe(ch *ch) {
```

Comment:
I think, the idea of having functions that use lock and functions that don't is creating more confusion rather than creating clarity. So, I have
1. Removed the `lock`, `unlock`, `*Unsafe` methods.
2. Directly access the mutex and map when multiple operations are performed while holding the lock on registry.
3. Use methods, that internally use mutex locks when only one operation is performed on the registry.

Your thoughts on this @matthiasgeihs @ggwpez ?

Reply:
This is not relevant now. Because, I have refactored the registry such that, the watcher uses only the methods defined on registry and all methods use the mutex lock.

See [the previous comment](https://github.com/hyperledger-labs/go-perun/pull/213#discussion_r721430298) for more details.